### PR TITLE
Update depositors (proxies, transfers) synchronously

### LINKED ISSUE #5457: Works deposited via proxy are always set to private visibility
### Descriptive summary

On nurax-pg, when depositing a work as a proxy, the visibility of the work is set to private even if the visibility on deposit was set to public.

### Expected behavior

Works deposited via proxy should reflect the visibility settings selected during the deposit process.

### Actual behavior

Works deposited via proxy are set to private.

### Steps to reproduce the behavior

1. Make sure that proxy deposit is enabled on nurax-pg and that you have an account set up as a proxy for another account
2. Navigate to Dashboard -> Works
3. Click "Add new work" and select "Generic Work"
4. Fill out required metadata and add a file
5. Select "Public" visibility
6. Select another user in the "Proxy depositors" box.
7. Check the Deposit agreement box and click "Save".
8. Observe message "Unauthorized The page you have tried to access is private" message

### PR DESCRIPTION
**Pull request by hackartisan**
supercedes #5537 
follow-up to #5529 and #5550
closes #5457 

Deprecates ContentDepositorChangeEventJob which called ChangeContentDepositorService, replacing them with a service that does the work and then backgrounds the event logging.

Follow-on work will move file set updates to a new background job, since there might be a lot of file sets. Issue to be created, linked here.

### CODE REVIEW

**Comment by hackartisan** [comment]
This should be in draft until #5550 is merged and rebased on that PR.

**Review by elrayle — CHANGES_REQUESTED** [review]
This is fantastic.  There is a lot in this PR and I appreciate all the time it took to get this right.

There is a request for one new test and a question about the order of steps for work_create transaction.

File: lib/hyrax/transactions/work_create.rb [hunk 1]
```diff
@@ -13,6 +13,7 @@ class WorkCreate < Transaction
                        'change_set.set_user_as_depositor',
                        'change_set.apply',
                        'work_resource.save_acl',
+                       'work_resource.change_depositor',
                        'work_resource.add_file_sets',
```
[the remark this answers is not in the document]
Comment (hackartisan) [reply lib/hyrax/transactions/work_create.rb:16]: Oh, good point with the order of the steps!! And I added the test.

**Review by elrayle — APPROVED** [review]
Thanks for reordering the work_create steps and getting in the test.  This was quite a slog, but well done.

**Pull request merged**